<launch>
  <!-- subscribes to cmd_vel topic -->
  <!-- publishes odom topic with base_link frame -->
  <!-- broadcasts odom->base_link transform -->
  <arg name="param" default="value" />
  <node pkg="roboteq_diff_driver" type="driver" name="roboteq_diff_driver" output="screen">
    <!-- enable broadcast of odom tf -->
    <param name="pub_odom_tf" value="true" />
    <!-- specify odom frame -->
    <param name="odom_frame" value="odom" />
    <!-- specify base frame -->
    <param name="base_frame" value="base_link" />
    <!-- specify cmd_vel topic -->
    <param name="cmdvel_topic" value="cmd_vel" />
    <!-- specify device for roboteq controller -->
    <param name="device" value="/dev/ttyACM0" />
    <!-- specify baudrate for roboteq controller -->
    <param name="baudrate" value="115200" />
    <!-- specify whether to use open-loop motor speed control (as opposed to closed-loop)-->
    <param name="open-loop" value="false" />
    <!-- specify robot wheel circumference in meters  -->
    <param name="wheel_circumference" value="0.3192" />
    <!-- specify robot track width in meters  -->
    <param name="track_wdith" value="0.4318" />
    <!-- specify pulse-per-revolution for motor encoders  -->
    <param name="encoder_ppr" value="900" />
    <!-- specify counts-per-revolution for motor encoders (ppr*4 for quadrature encoders)  -->
    <param name="encoder_cpr" value="3600" />
  </node>
</launch>
